Source/WebKit/UIProcess/WebFrameProxy.cpp

```diff
@@ -495,6 +495,50 @@ void WebFrameProxy::didCreateSubframe(WebCore::FrameIdentifier frameID, String&&
     m_childFrames.add(WTFMove(child));
 }
 
+void WebFrameProxy::createLocalFrameInProcessWithEffectiveOrigin(WebProcessProxy& newProcess, SecurityOriginData& effectiveOrigin, API::Navigation& navigation, BrowsingContextGroup& group)
+{
+    // Creates a local frame in a given process and with a given origin.
+    // This will create a provisional frame and immediately commit it, so it
+    // is meant for sites with no network resources to load (i.e. about:blank).
+
+    RefPtr page = m_page.get();
+    Site mainFrameSite(page->mainFrame()->url());
+    auto mainFrameDomain = mainFrameSite.domain();
+
+    // Store the previous process this WebFrame was running in so in didCreateFrameImmediately
+    // we can notify it that it's LocalFrame should be converted to a RemoteFrame.
+    m_processBeforeImmediateLoad = RefPtr { this->process() };
+
+    // Store the new process for this frame so we can update WebFrameProxy
+    // to point to it later.
+    m_frameProcessAfterImmediateLoad = RefPtr { group.ensureProcessForSite(Site { effectiveOrigin }, mainFrameSite, newProcess, page->protectedPreferences(), InjectBrowsingContextIntoProcess::No) };
+
+    newProcess.send(Messages::WebFrame::CreateProvisionalFrameAndCommit(ProvisionalFrameCreationParameters {
+        frameID(),
+        std::nullopt,
+        layerHostingContextIdentifier(),
+        effectiveSandboxFlags(),
+        effectiveReferrerPolicy(),
+        scrollingMode(),
+        remoteFrameSize(),
+        }), frameID());
+}
+
+void WebFrameProxy::didCreateFrameImmediately(FrameIdentifier frameID)
+{
+    // Point this WebFrameProxy to the new process for this frame
+    if (m_frameProcessAfterImmediateLoad) {
+        setProcess(*m_frameProcessAfterImmediateLoad);
+        m_frameProcessAfterImmediateLoad = nullptr;
+    }
+
+    // Notify the old process that it's LocalFrame has been loaded in a different process
+    if (m_processBeforeImmediateLoad) {
+        m_processBeforeImmediateLoad->send(Messages::WebPage::LoadDidCommitInAnotherProcess(frameID, m_layerHostingContextIdentifier), protectedPage()->webPageIDInProcess(*m_processBeforeImmediateLoad));
```

Comment:
This probably doesn't need to wait for the IPC roundtrip of CreateProvisionalFrameAndCommit to send.  Would it work to send it to the other process at the same time?